File: elixir/acronym.ex
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split([" ", "-"])
    |> Enum.map(fn s ->
      s |> String.replace(~r/\_/, "") |> String.first()
    end)
    |> Enum.join()
    |> String.upcase()
  end
end

File: dart/acronym.dart
class Acronym {
  String abbreviate(String s) {
    return s
        .replaceAll('-', ' ')
        .split(' ')
        .where((w) => w.trim().isNotEmpty)
        .map((w) => w.replaceAll(RegExp(r'[^a-zA-Z]'), '').toUpperCase()[0])
        .join();
  }
}
